flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotProvider.java

```diff
@@ -62,11 +51,13 @@ default void start(ComponentMainThreadExecutor mainThreadExecutor) {
 	 * @param allocationTimeout after which the allocation fails with a timeout exception
 	 * @return The future of the allocation
 	 */
-	CompletableFuture<LogicalSlot> allocateSlot(
+	default CompletableFuture<LogicalSlot> allocateSlot(
 		SlotRequestId slotRequestId,
 		ScheduledUnit scheduledUnit,
 		SlotProfile slotProfile,
-		Time allocationTimeout);
+		Time allocationTimeout) {
```

Comment:
Looks to me default body is not needed. I tried removing the default body and flink-runtime still compiles.
Correct me if I miss anything.